microcosm_pubsub/tests/test_consumer.py

```diff
@@ -119,28 +125,29 @@ def test_json_consume():
     Test that message sent as JSON could be delivered
 
     """
-    graph = create_daemon_with_naive_cache()
-    # simulate the response structure
-    # replacing MagicMock with real reader
-    graph.sqs_consumer.sqs_client = SQSJsonReader(dict(
+    event = dict(
             MessageId=MESSAGE_ID,
             ReceiptHandle=RECEIPT_HANDLE,
             MD5OfBody="7efaa8404863d47c51ed0e20b9014aec",
             Body=dumps(dict(
                 data="data",
                 mediaType=DerivedSchema.MEDIA_TYPE,
             )),
-        ),
-    )
+        )
 
+    graph = create_daemon_with_naive_cache(event)
+    # simulate the response structure
+    # replacing MagicMock with real reader
+    graph.sqs_consumer.sqs_client = SQSJsonReader(event)
     messages = graph.sqs_consumer.consume()
 
     # and response translated properly
+    assert_that(graph.sqs_consumer, is_(instance_of(SQSConsumer)))
+    assert_that(graph.sqs_envelope, is_(instance_of(RawSQSEnvelope)))
     assert_that(messages, has_length(1))
     assert_that(messages[0].consumer, is_(equal_to(graph.sqs_consumer)))
-    assert_that(messages[0].message_id, is_(equal_to(MESSAGE_ID)))
     assert_that(messages[0].receipt_handle, is_(equal_to(RECEIPT_HANDLE)))
     assert_that(messages[0].content, is_(equal_to(dict(
         data="data",
-        media_type=DerivedSchema.MEDIA_TYPE,
+        mediaType=DerivedSchema.MEDIA_TYPE,
```

Comment:
Is this casing change due to the envelope change?

Reply:
The worry would be when we switch default behaviour to the previous lambda type with `media_type` any of them that had any hard-coded assumptions about the message structure would break. Consider it breaking a contract in the message structure that something might rely on.

From a quick search of our consumers this doesn't seem to be the case, the only place we directly refer to `mediaType` is in this library and tests. Though that wouldn't be ideal if switching to the new consumer type broke 100s of tests across a lot of repos ;)